PerformanceAnalyzerGQI/GetPerformanceMetricsMultiFile.cs

```diff
@@ -0,0 +1,268 @@
+				sw = Stopwatch.StartNew();
+				var gqiRows = new GQIPage(filteredRows.Select(x => CreateGqiRow(x)).ToArray()) { HasNextPage = false };
+				logger.Information($"Creating GQI rows done in - {sw.ElapsedMilliseconds}ms.");
+
+				return gqiRows;
+			}
+			catch (Exception ex)
+			{
+				logger.Error(ex.ToString());
+			}
+
+			return new GQIPage(new GQIRow[0]);
+		}
+
+		private void LoadFiles(ConcurrentBag<FileMetricRow> rows)
+		{
+			var fileInfos = GetFilesToLoad();
+
+			Parallel.ForEach(fileInfos, fileInfo =>
+			{
+				if (fileInfo.LastWriteTime > startTimeFilter && fileInfo.LastWriteTime < endTimeFilter)
+				{
+					using (StreamReader reader = new StreamReader(fileInfo.FullName))
+					using (JsonTextReader jsonReader = new JsonTextReader(reader))
+					{
+						JsonSerializer serializer = new JsonSerializer();
+						var deserializedLogs = serializer.Deserialize<ConcurrentBag<PerformanceLog>>(jsonReader);
+						ProcessLogs(rows, fileInfo, deserializedLogs);
+					}
+				}
+			});
+		}
+
+		private IEnumerable<FileInfo> GetFilesToLoad()
+		{
+			DirectoryInfo directoryInfo = new DirectoryInfo(fileLocation);
+			FileInfo[] fileInfos = directoryInfo.GetFiles(searchPattern, searchOption);
+
+			long currentTotalSize = 0;
+			var fileInfosFiltered = fileInfos.OrderByDescending(x => x.LastWriteTime).TakeWhile(x =>
+			{
+				currentTotalSize += x.Length;
+				return currentTotalSize < _totalFileSizeLimit;
+			});
+
+			return fileInfosFiltered;
+		}
+
+		private void ProcessLogs(ConcurrentBag<FileMetricRow> rows, FileInfo fileInfo, ConcurrentBag<PerformanceLog> deserializedLogs)
+		{
+			foreach (PerformanceLog performanceLog in deserializedLogs.Where(x => x.StartTime > startTimeFilter && x.StartTime < endTimeFilter))
+			{
+				foreach (PerformanceData performanceData in performanceLog.Data)
+				{
+					ProcessSubMethods(performanceData, rows, 0, fileInfo.Name);
+				}
+			}
+		}
+
+		private void ProcessSubMethods(PerformanceData data, ConcurrentBag<FileMetricRow> rows, int level, string fileName)
+		{
+			if (data == null)
+			{
+				return;
+			}
+
+			rows.Add(CreateFileMetricRow(data, level, fileName));
+
+			if (data.SubMethods != null && data.SubMethods.Any())
+			{
+				foreach (var subMethod in data.SubMethods)
+				{
+					ProcessSubMethods(subMethod, rows, level + 1, fileName);
+				}
+			}
+		}
+
+		private FileMetricRow CreateFileMetricRow(PerformanceData performanceData, int level, string fileName)
+		{
+			return new FileMetricRow
+			{
+				FileName = fileName,
+				ClassName = performanceData.ClassName,
+				MethodName = performanceData.MethodName,
+				StartTime = performanceData.StartTime,
+				EndTime = performanceData.StartTime + performanceData.ExecutionTime,
+				ExecutionTime = performanceData.ExecutionTime,
+				MethodLevel = level,
+				Metadata = performanceData.ShouldSerializeMetadata() ? JsonConvert.SerializeObject(performanceData.Metadata) : string.Empty,
```

Comment:
Can we avoid deserializing and serializing again? Might potentially save us a lot of serialization/deserialization. Maybe using a custom deserializer for the PerformanceLog class?

Reply:
Initial tests didn't produce noticeable improvements so agreed to skip this for now. It can be reevaluated in a future version.